dss/stepfunctions/__init__.py

```diff
@@ -16,18 +54,62 @@ def step_functions_invoke(state_machine_name_template: str, execution_name: str,
     """
 
     sfn = boto3.client('stepfunctions')
-    sts_client = boto3.client("sts")
 
     execution_input = json.dumps(input)
+    state_machine_arn = step_functions_arn(state_machine_name_template)
 
-    region = boto3.Session().region_name
-    accountid = sts_client.get_caller_identity()['Account']
-    stage = os.environ["DSS_DEPLOYMENT_STAGE"]
-    sfn_name = state_machine_name_template.format(stage=stage)
-    state_machine_arn = f"arn:aws:states:{region}:{accountid}:stateMachine:{sfn_name}"
     response = sfn.start_execution(
         stateMachineArn=state_machine_arn,
         name=execution_name,
         input=execution_input
     )
     return response
+
+
+def step_functions_describe_execution(state_machine_name_template: str, execution_name: str) -> typing.Any:
+    """
+    Return description of a step function exectuion, possible in-progress, completed, errored, etc.
+    :param state_machine_name_template:
+    :param execution_name:
+    :return:
+    """
+
+    sfn = boto3.client('stepfunctions')
+
+    execution_arn = step_functions_execution_arn(state_machine_name_template, execution_name)
+
+    resp = sfn.describe_execution(
+        executionArn=execution_arn
+    )
+
+    return resp
+
+
+def step_functions_list_executions(state_machine_name_template: str, k_results_per_page: int=100) -> typing.Iterable:
+    """
+    List step function executions, peforming paging in the background.
+    Maximum 100 results per page.
+    :param state_machine_name_template:
+    :param k_results_per_page:
+    :return:
+    """
+
+    sfn = boto3.client('stepfunctions')
+
+    state_machine_arn = step_functions_arn(state_machine_name_template)
+
+    kwargs = {
+        'stateMachineArn': state_machine_arn,
+        'maxResults': k_results_per_page
+    }
```

Comment:
Nit: I usually initialize kwargs using the `dict` constructor. The keys are bound to be valid Python identifiers and more closely resembles a direct invocation, without `**`.